ci-operator/step-registry/azure/provision/bastionhost/azure-provision-bastionhost-commands.sh

```
@@ -204,8 +204,110 @@ if [[ -z "${bastion_subnet}" ]] && [[ -z "${bastion_nsg}" ]]; then
+            echo "ERROR: Could not query VNET address prefix" && exit 1
+        fi
+        echo "Found VNET address prefix: ${AZURE_VNET_ADDRESS_PREFIXES}"
+    fi
+
+    # Find an available /24 subnet within the VNET
+    echo "Querying existing subnets to find available address space..."
+    existing_subnets=$(az network vnet subnet list --resource-group ${bastion_rg} --vnet-name ${bastion_vnet_name} --query '[].addressPrefix' -o tsv)
+    echo "Existing subnets: ${existing_subnets}"
+
+    # Extract the first two octets from VNET prefix (e.g., 10.0 from 10.0.0.0/16)
+    vnet_base=$(echo ${AZURE_VNET_ADDRESS_PREFIXES%/*} | awk -F'.' '{print $1"."$2}')
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)